gadget/update.go

```diff
@@ -258,9 +265,52 @@ func EnsureLayoutCompatibility(gadgetLayout *LaidOutVolume, diskLayout *OnDiskVo
 		// size and offset, so the last thing to check is that the filesystem
 		// matches (or that we don't care about the filesystem).
 
-		// TODO: here we need to handle in the strict case partitions which are
-		// to be created at install and are encrypted like ubuntu-data as they
-		// will not match filesystems exactly and need some massaging
+		// first handle the strict case where this partition was created at
+		// install in case it is an encrypted one
+		if opts.AssumeCreatablePartitionsCreated && IsCreatableAtInstall(gv) {
+			// only partitions that are creatable at install can be encrypted,
+			// check if this partition was encrypted
+			if encTypeParams, ok := opts.ExpectedStructureEncryption[gs.Name]; ok {
+				// "method" is a required and expected key in the parameters map
+				encMethod, ok := encTypeParams["method"]
```

Comment:
I don't have a strong opinion but I'm not a fan of too many string constants. In the past we had issues with constants that ended up incorporating a typo, this can happen though with strings too with editor autocompletion.  In general it's a good idea to use direct string at least somewhere in tests if the code is using constants.

Reply:
Using a struct as suggested makes this problem go away